### PR TITLE
Add spell effect for Life Link

### PR DESCRIPTION
**Pull request by ammalagonc**
Add spell effect for the Life Link Oracle spell

### CODE REVIEW

**Comment by simonward** [comment]
Consider making the inline rolls auto-heighten.

File: packs/spell-effects/spell-effect-life-link.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                    "life-linked",
+                    {
+                        "or": [
+                            "self:type:character",
+                            "self:type:npc"
+                        ]
+                    }
+                ],
+                "value": {
+                    "brackets": [
```
Comment (simonward) [inline packs/spell-effects/spell-effect-life-link.json:51]: Favor `ternary()` over brackets where possible.

File: packs/spell-effects/spell-effect-life-link.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            "license": "OGL",
+            "remaster": false,
+            "title": "Pathfinder Advanced Player's Guide"
+        },
+        "rules": [
+            {
+                "alwaysActive": false,
+                "domain": "all",
+                "key": "RollOption",
+                "label": "Life Link",
```
[the remark this answers is not in the document]
Comment (ammalagonc) [reply packs/spell-effects/spell-effect-life-link.json:28]: The way this works - the effect generates a toggle that allows the user/GM to block the first instance of damage and then un-toggle it to allow them to continue to apply damage as normal.

File: packs/spell-effects/spell-effect-life-link.json [hunk 3]
```diff
@@ -0,0 +1,87 @@
+{
+    "_id": "8SkrekdJ9h3T664t",
+    "img": "systems/pf2e/icons/spells/life-link.webp",
+    "name": "Spell Effect: Life Link",
+    "system": {
+        "description": {
+            "value": "<p>Granted by @UUID[Compendium.pf2e.spells-srd.Item.Life Link]</p>\n<p>You forge a connection of life energy between you and your target, distributing harm between both of you.<br /><br /><strong>Apply Damage (Caster)</strong> @Damage[(3)[Hit Points]].</p>\n<p><strong>Heightened (3rd)</strong> @Damage[(5)[Hit Points]].</p>\n<p><strong>Heightened (6th)</strong>@Damage[(10)[Hit Points]].</p>\n<p><strong>Heightened (9th)</strong>@Damage[(15)[Hit Points]].</p>"
```
Comment (TikaelSol) [inline packs/spell-effects/spell-effect-life-link.json:7]: We don't use `<br>` tags for newlines. Effects should also say what they do and nothing else.

File: packs/spell-effects/spell-effect-life-link.json [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                    {
+                        "not": "self:signature:{item|origin.signature}"
+                    }
+                ],
+                "toggleable": true
+            },
+            {
+                "key": "ActiveEffectLike",
+                "mode": "override",
+                "path": "system.attributes.hardness.value",
```
Comment (TikaelSol) [inline packs/spell-effects/spell-effect-life-link.json:40]: Sorry, but this isn't a method we will use to implement this. Life link isn't hardness.

**Comment by TikaelSol** [comment]
Changed the title to be in imperative case.

File: packs/spell-effects/spell-effect-life-link.json (hunk 4 above)
Comment (ammalagonc) [reply packs/spell-effects/spell-effect-life-link.json:40]: Fair enough - Do you want me to close this PR then?

**Pull request closed without merging**